src/coreclr/nativeaot/Runtime/EHHelpers.cpp

```diff
@@ -490,7 +487,7 @@ LONG WINAPI RhpVectoredExceptionHandler(PEXCEPTION_POINTERS pExPtrs)
 
             // Do not use ASSERT_UNCONDITIONALLY here. It will crash because of it consumes too much stack.
             PalPrintFatalError("\nProcess is terminating due to StackOverflowException.\n");
-            RaiseFailFastException(pExPtrs->ExceptionRecord, pExPtrs->ContextRecord, 0);
+            PalRaiseFailFastException(pExPtrs->ExceptionRecord, pExPtrs->ContextRecord, 0);
```

Comment:
This is Windows-specific code calling Windows-specific API. There is no point going through a PAL layer in cases like these.

Reply:
I see. The problem with this model is there is an odd consistency in files. It isn't obvious/clear on when we can avoid the PAL. If this file was clearly in a "windows" directory it would be obvious. In thics case though we have 100s of lines of code that defined in complex ways that make it tricky to know. If we instead just implement a PAL consistently we can use the simple rule of "only use PAL APIs". I'll change it, but I don't think this is a helpful way of writing code in a file that isn't defined as being platform specific.

See the other "Pal" functions in the same block.